lib/src/cli.rs

```diff
@@ -281,17 +285,39 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
     let commit = booted_deployment.csum();
     let state = ostree_container::store::query_image_commit(repo, &commit)?;
     let digest = state.manifest_digest.as_str();
-    let fetched = pull(repo, &imgref, opts.quiet).await?;
 
-    if fetched.merge_commit.as_str() == commit.as_str() {
-        println!("Already queued: {digest}");
+    if opts.check {
+        // pull the image manifest without the layers
+        let config = Default::default();
+        let mut imp = ostree_container::store::ImageImporter::new(repo, &imgref, config).await?;
+        match imp.prepare().await? {
+            PrepareResult::AlreadyPresent(c) => {
+                println!(
+                    "No changes available for {}. Latest digest: {}",
+                    imgref, c.manifest_digest
+                );
+            }
+            PrepareResult::Ready(p) => {
+                println!(
+                    "New manifest available for {}. Digest {}",
```

Comment:
One less than ideal thing here is that it leaves open the problem of how programmatic access would work - if I'm scripting bootc, I'd have to parse this text designed for human readability.

We already have a `--touch-if-changed` CLI option which is designed for the non-check case, so let's honor it with `--check` too.